pkg/chrootarchive/archive.go

```diff
@@ -83,6 +83,18 @@ func untarHandler(tarArchive io.Reader, dest string, options *archive.TarOptions
 		}
 	}
 
+	destVal, err := newUnpackDestination(root, dest)
+	if err != nil {
+		return err
+	}
+	closeDestVal := true
+	defer func() {
```

Comment:
Passing the FD to `Exec` does not close it AFAICS, so this should be closed unconditionally.

Reply:
Right, thanks!